qa/rpc-tests/test_framework/util.py

```diff
@@ -352,6 +356,28 @@ def start_node(i, dirname, extra_args=None, rpchost=None, timewait=None, binary=
 
     return proxy
 
+def assert_start_raises_init_error(i, dirname, extra_args=None, expected_msg=None):
+    log_stderr = tempfile.SpooledTemporaryFile(max_size=2**16)
```

Comment:
Consider opening the file using the `with log_stderr as...` form. That means the file will be closed automatically (even if an exception is raised on the way), and you won't have to close it yourself with the `finally` branch at the bottom.